src/jdk.incubator.vector/share/classes/jdk/incubator/vector/ByteVector.java

```diff
@@ -625,34 +641,15 @@ ByteVector lanewise(VectorOperators.Binary op,
     ByteVector lanewise(VectorOperators.Binary op,
                                   Vector<Byte> v,
                                   VectorMask<Byte> m) {
-        if (op == ROR || op == ROL) {
-            return blend(lanewise(op, v), m);
-        }
-
         ByteVector that = (ByteVector) v;
         that.check(this);
-        if (op == DIV) {
-            VectorMask<Byte> eqz = that.eq((byte)0);
-            if (eqz.and(m).anyTrue()) {
-                throw that.divZeroException();
-            }
-            // suppress div/0 exceptions in unset lanes
-            that = that.lanewise(NOT, eqz);
+        VectorSpecies<Byte> maskSpecies =
+            ((jdk.incubator.vector.VectorMask<Byte>) m).vectorSpecies();
+        if (maskSpecies != vspecies()) {
+            throw AbstractSpecies.checkFailed(maskSpecies, vspecies());
```

Comment:
@iwanowww @PaulSandoz , the new commit add a type check to the mask value in the masked `lanewise` API. This could raise an implicit null check for the mask value. I tried to use `m.getClass()` at first, but I didn't find a reasonable way to use it in this API. Does this change make sense for you? Thanks!

Reply:
Yes, it looks fine to me. 

I suggest to factor the check into a helper method (`AbstractVector::check(VectorMask m)`) and call it next to the check on `Vector`.

Reply:
Thanks! I will update the codes according to your suggestion!